cmd/cpu-node-labeller/cpu-node-labeller.go

```diff
@@ -78,13 +67,42 @@ func main() {
 	}
 
 	node.RemoveCPUModelNodeLabels(n)
-	node.AddNodeLabels(n, cpuModels)
+	node.AddNodeLabels(n, features)
 
 	err = node.UpdateNode(cli, n)
 	if err != nil {
 		glog.Fatalf("error while updating node: %s", err)
 		os.Exit(1)
 	}
 
-	glog.Info("updated node!")
+	glog.Info("node updated!")
+}
+
+func runFile(fileDir, fileName string, features map[string]string) error {
+	path := filepath.Join(fileDir, fileName)
+	filestat, err := os.Stat(path)
+	if err != nil {
+		return err
+	}
+
+	//if file is regular and executable
+	if filestat.Mode().IsRegular() && filestat.Mode()&0111 != 0 {
+		cmd := exec.Command(path)
+		var stdout, stderr bytes.Buffer
+		cmd.Stdout = &stdout
+		cmd.Stderr = &stderr
+
+		// Run hook
+		err = cmd.Run()
+
+		lines := bytes.Split(stdout.Bytes(), []byte("\n"))
+		for _, feature := range lines {
+			if len(feature) == 0 {
+				continue
+			}
+
+			features[string(feature)] = "true"
```

Comment:
OK for the time being. In the future we may want to allow negative labels.

Reply:
In the future, we will be using openshift NFD :)

Reply:
Negative labels could be represented by not adding the label at all :)